//   Copyright 2016 Pivotal Software, Inc.
//
//   Licensed under the Apache License, Version 2.0 (the "License");
//   you may not use this file except in compliance with the License.
//   You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//   Unless required by applicable law or agreed to in writing, software
//   distributed under the License is distributed on an "AS IS" BASIS,
//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//   See the License for the specific language governing permissions and
//   limitations under the License.

syntax = "proto3";

package quickstep;

service DataExchange {
  rpc Pull (PullRequest) returns (PullResponse) {}
}

message PullRequest {
  fixed64 block_id = 1;
}

message PullResponse {
  bool is_valid = 1;
  uint64 num_slots = 2;
  bytes block = 3;
}
